### DISCUSSION TITLE
How do I return a different object?

### DISCUSSION THREAD

**Discussion by Ethan-Hill** [nextauthjs/next-auth#937] [Help] [answered]
Hey, so I'm currently using the discord OAuth provider and I was wanting to know how I could change the returned object to the session.

I'm looking to return the profile object instead of the user object or even better have access to both in the session.

Any help would be appreciated, thanks.

**Comment by balazsorban44** [comment, marked as answer]
Hi there! Check out the `jwt` and `session` callbacks:

https://next-auth.js.org/configuration/callbacks

**Reply by nodgear** [reply]
Okay, @balazsorban44 .
The problem is: Session callback returns User object instead of token object, making it harder to get said data into the session context.
```js
"jwt: async(token, user, account, profile, isNewUser) => {
      if (account?.accessToken) {
        console.log("SHIT IS VALID")
        token.id = profile.id;
        token.accessToken = profile.accessToken;
        token.tokenExpiration = profile.accessToken;
        token.refreshToken = profile.refreshToken
      }
      // token.id = account.id;
      return token
    }
```

The code above being my attempt to storing stuff on token
How could we recover it from session, while second argument is user instead?

```js
     session: async (session, user) => {
      session.user.id = user.sub
      console.log("RAN SESSION", session, user)
      return session
    },
      // Outputs user object:
      // RAN SESSION {
      // user: {
      // name: '****Nodge',
      // email: '********',
      // image: '*****',
      // id: '****'
     //  },
     //  expires: '2021-05-11T09:27:22.836Z'
     //  } 
    ```
